model/manifest.go

```diff
@@ -196,9 +196,21 @@ type Manifest struct {
 }
 
 type ManifestServer struct {
-	// Executables are the paths to your executable binaries, specifying multiple entry points
-	// for different platforms when bundled together in a single plugin.
-	Executables *ManifestExecutables `json:"executables,omitempty" yaml:"executables,omitempty"`
+	// AllExecutables are the paths to your executable binaries, specifying multiple entry
+	// points for different platforms when bundled together in a single plugin.
+	AllExecutables map[string]string `json:"executables,omitempty" yaml:"executables,omitempty"`
+
+	// Executables is a legacy field populated with a subset of supported platform executables.
+	// When unmarshalling, Executables is authoritative for the platform executable paths it
+	// contains, overriding any values in AllExecutables. When marshalling, AllExecutables
+	// is authoritative.
+	//
+	// Code duplication is avoided when (un)marshalling by leveraging type aliases in the
+	// various (Un)Marshal(JSON|YAML) methods, since aliases don't inherit the aliased type's
+	// methods.
+	//
+	// In v6.0, we should remove this field and rename AllExecutables back to Executables.
```

Comment:
Ticket for this?

Reply:
https://mattermost.atlassian.net/browse/MM-27593